### ISSUE TITLE
Psychopy will not launch

### ISSUE THREAD

**Issue by sarah756** [psychopy/psychopy#2131]
Hi, I am struggling to open the latest version of Psychopy, as it will not launch, I have followed the instructions to try and resolve this issue but nothing seems to work. I was wondering if there is any solution to this?

Many thanks.

**Comment by peircej** [comment]
From that information I can't help. The troubleshooting guide tells you how to try and get an error message. That will help us work out what's going wrong.

Also though, could you tell us what operating system and confirm that by "latest" do you mean 3.0.0b12?

**Comment by peircej** [comment]
http://www.psychopy.org/troubleshooting.html

**Comment by sarah756** [comment]
I am using the 3.0.0b12 version, on a Mac. When I start Psychopy it either gives me the option to 'terminate' or 'open console'.

**Comment by peircej** [comment]
OK, well, assuming you've put the app in the Applications folder, could you:

1. go to the terminal program (quickest way it to do Command-Space and start typing `terminal`)
2. paste /Applications/PsychoPy3.app/Contents/MacOS/python -m psychopy.app.psychopyApp

hopefully that way you'll see an error message there

**Comment by hoechenberger** [comment]
@sarah756 Any news on this one?

**Comment by sarah756** [comment]
Hi, sorry. Yes, it's working fine now. Thank you both for your help.

**Comment by hoechenberger** [comment]
@sarah756 Could you let us know what was causing the issue? :)

**Comment by sarah756** [comment]
Hi, I had to move the software to the applications folder on the Mac :)

**Comment by aelreym** [comment]
Hi, I have the same issue and the error message in the terminal is the following: /Applications/PsychoPy3.app/Contents/MacOS/python: Error while finding module specification for 'psychopy.app.psychopyApp' (ModuleNotFoundError: No module named 'psychopy')
What should I do to resolve this? 
Thanks.

**Comment by peircej** [comment]
Not sure what's going on with just a small number of macs that don't launch. Is this with the new 2020.1.1? What version of MacOS?
I'm assuming you clicked on the security settings to "Open anyway". If you are trying the new 2020.1.1 then I wonder if you would check/upload the contents of ~/.psychopy3/last_app_load.log to see if there's any useful clue there

**Comment by aelreym** [comment]
I have tried to reinstall psychopy - also with pip and anaconda, but it's still the same, an error message and open console or terminate. 

Here the answers I get in the terminal: 



**leareymann@Leas-Air ~ % /Applications/PsychoPy3.app/Contents/MacOS/python -m psychopy.app.psychopyApp**

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/psychopy/__init__.py", line 44, in <module>
    from psychopy.preferences import prefs
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/psychopy/preferences/__init__.py", line 13, in <module>
    from . import preferences as prefsLib
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/psychopy/preferences/preferences.py", line 9, in <module>
    import platform
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/platform.py", line 116, in <module>
    import sys, os, re, subprocess
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 136, in <module>
    import _posixsubprocess
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/lib-dynload/_posixsubprocess.cpython-36m-darwin.so, 2): Symbol not found: __Py_set_inheritable_async_safe
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/lib-dynload/_posixsubprocess.cpython-36m-darwin.so
  Expected in: flat namespace
 in /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/lib-dynload/_posixsubprocess.cpython-36m-darwin.so
leareymann@Leas-Air ~ % **/.psychopy3/last_app_load.log** 
zsh: no such file or directory: /.psychopy3/last_app_load.log

**Comment by aelreym** [comment]
It is with the new version of Psychopy and MacOS 10.15.1

**Comment by hoechenberger** [comment]
> **leareymann@Leas-Air ~ % /Applications/PsychoPy3.app/Contents/MacOS/python -m psychopy.app.psychopyApp**
> 
> Traceback (most recent call last):
> File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 183, in _run_module_as_main

This seems to be trying to run a different (manual) PsychoPy installation, not the one that ships with the standalone package…

**Comment by aelreym** [comment]
I don't know what to do to solve this issue. Can you help?

**Comment by peircej** [comment]
I have an idea. I bet it's that platform isn't being provided in the Standalone so it's gone looking elsewhere and found another (incompatible) version on the operating system. I'll repackage with `platform` included.
Looks like the (new) system for logging while loading the app is already useful! :-) 
Thanks for your help here @aelreym

**Comment by aelreym** [comment]
python3 -m pip install repackage psychopy --only-binary=:all --platform macosx_10_15_1 -- python-version 36 --abi cp36
ERROR: When restricting platform and interpreter constraints using --python-version, --platform, --abi, or --implementation, either --no-deps must be set, or --only-binary=:all: must be set and --no-binary must not be set (or must be set to :none:).

**Comment by aelreym** [comment]
I have tried to repackage with platform included:
 ~ % python3 -m pip install repackage psychopy --platform

Usage:   
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 -m pip install [options] <requirement specifier> [package-index-options] ...
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 -m pip install [options] -r <requirements file> [package-index-options] ...
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 -m pip install [options] [-e] <vcs project url> ...
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 -m pip install [options] [-e] <local project path> ...
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 -m pip install [options] <archive url/path> ...

--platform option requires 1 argument


What is this argument needed?

**Comment by peircej** [comment]
I misinterpreted the error when I looked yesterday - I thought it had just fetched the wrong copy of `platform` but actually its the entire wrong import.

I think the issue is coming from 2 things: 
1. certainly you've got multiple installations of PsychoPy (1 is standalone but you've installed conda and pip versions too and they're competing). In general that's fine.
2. for some reason the standalone installation is then using the other installation to import and some library versions are apparently incompatible

I'm not sure why (2) is occuring - why the Standalone version that you're trying to launch is detecting the version in /Library/Frameworks. Have you altered your path variable in PsychoPy? Maybe you've added something to your psychopy prefs to include site-packages from your LibraryFrameworks (e.g. to make use of another lib?)

Anyway, I'm really thinking this is something about your installation rather than a bug in PsychoPy so I'm closing it again as an issue
